lib/parser/token-parsers.h

```diff
@@ -289,7 +289,7 @@ struct BOZLiteral {
     space.Parse(state);
     const char *start{state.GetLocation()};
     std::optional<const char *> at{nextCh.Parse(state)};
-    if (!at.has_value()) {
+    if (!at) {
```

Comment:
I'm seeing a bunch of compilation errors with GCC 8.3.0. For example, on this line:
```
../../lib/parser/token-parsers.h:292:9: error: could not convert 'Fortran::parser::operator!<std::optional<const char*> >(at)' from 'Fortran::parser::NegatedParser<std::optional<const char*> >' to 'bool'
```
It doesn't complain about line 143, which looks the same.

Reply:
Already fixed; will not merge until all builds and tests are clean.

Reply:
What was wrong? It looks okay to me.

Reply:
`operator!()` was already overloaded in the context of parser combinators.